Update Docs: Intro “start-with-a-template”

surjithctly opened the pull request:
Automatically generated by Netlify CMS

erezrokah reviewed (changes requested):
Thanks @surjithctly and sorry for the late reply.
Great initiative, just a few comments:
1. Since you're using `stack=cms` in the deploy link should we change the backend name to `git-gateway`:
https://github.com/surjithctly/neat-starter/blob/71bdcee2ea58709bb951e774d880036c9e32ff65/admin/config.yml#L2
2. I tried deploying via the deploy button and this resulted in a css file missing:
![image](https://user-images.githubusercontent.com/26760571/94447418-67126f80-01b2-11eb-8e64-7d316e960a94.png)
changing the default build command from `eleventy` (set by Netlify) to `npm run build` seems to fix this. Should we add a `netlify.toml` file to fix this?
3. None of the navigation buttons work:
![image](https://user-images.githubusercontent.com/26760571/94447859-f455c400-01b2-11eb-87b9-692b7a307f9c.png)
Is that expected?
4. Since only a file collection is configured users can't add content to the site via the CMS:
https://github.com/surjithctly/neat-starter/blob/71bdcee2ea58709bb951e774d880036c9e32ff65/admin/config.yml#L13
should we add a folder collection too?
5. It would be nice to link to a published version of the site in the readme so people can experiment without deploying it first.

surjithctly commented:
Hello @erezrokah 

Thanks for the review. 

1. I honestly don't know what `stack=cms` means or what are the other options, I just copied from the other starter. 
Anyway, I have changed to `git-gateway`. Any docs link to explain it better?

2. Added `netlify.toml` and one click deploy should work fine. 

3. Navigation menu is added for illustration only, I made the text naming to avoid confusions. 

4. Done. Added blog folder collections

5. I don't know if you noticed, there is a link in the sidebar (github changed from top to sidebar, now easy to miss) 
  Anyway, I have also added it in the readme as well. 

Hope all good now for merging.

erezrokah commented:
> 1. I honestly don't know what `stack=cms` means or what are the other options, I just copied from the other starter.
>    Anyway, I have changed to `git-gateway`. Any docs link to explain it better?

Thank you for fixing those issues. Nice work. I've opened an issue for our docs team to handle that since I couldn't find where it is documented.

I tried the deploy button again and I'm getting a build error:
https://app.netlify.com/sites/silly-clarke-9b26e2/deploys/5f83162e9559f97ecc1b5926
```
4:27:20 PM: $ npm run build
4:27:21 PM: > @ build /opt/build/repo
4:27:21 PM: > cross-env NODE_ENV=production eleventy && cross-env NODE_ENV=production postcss static/css/tailwind.css --o _site/static/css/style.css
4:27:21 PM: Problem writing Eleventy templates: (more in DEBUG output)
4:27:21 PM: > Having trouble rendering html template ./index.html
4:27:21 PM: `TemplateContentRenderError` was thrown
4:27:21 PM: > (./index.html)
4:27:21 PM:   Error: template not found: _includes/partials/content.njk
4:27:21 PM: `Template render error` was thrown:
4:27:21 PM:     Template render error: (./index.html)
4:27:21 PM:       Error: template not found: _includes/partials/content.njk
4:27:21 PM:         at Object._prettifyError (/opt/build/repo/node_modules/nunjucks/src/lib.js:36:11)
4:27:21 PM:         at /opt/build/repo/node_modules/nunjucks/src/environment.js:561:19
4:27:21 PM:         at eval (eval at _compile (/opt/build/repo/node_modules/nunjucks/src/environment.js:631:18), <anonymous>:14:11)
4:27:21 PM:         at createTemplate (/opt/build/repo/node_modules/nunjucks/src/environment.js:293:11)
4:27:21 PM:         at next (/opt/build/repo/node_modules/nunjucks/src/lib.js:328:7)
4:27:21 PM:         at handle (/opt/build/repo/node_modules/nunjucks/src/environment.js:327:11)
4:27:21 PM:         at /opt/build/repo/node_modules/nunjucks/src/environment.js:337:9
4:27:21 PM:         at next (/opt/build/repo/node_modules/nunjucks/src/lib.js:326:7)
4:27:21 PM:         at Object.asyncIter (/opt/build/repo/node_modules/nunjucks/src/lib.js:332:3)
```

surjithctly commented:
@erezrokah Fixed. Made few changes because you were late to respond :) forgot to push new changes. 

All good now.

erezrokah reviewed (changes requested):
Thank you for following up so quickly.
I've tried the button and it deploys successfully 🎉 
However, the Identity flow doesn't work - I got an email inviting me to join, but the accept link is not handled in the site.
You can add a reference to the Netlify Identity Widget to handle it. See https://github.com/ixartz/Eleventy-Starter-Boilerplate/blob/9c39192352531516e6d0e8c489ba64cdf0f07dfb/src/_includes/layouts/base.ejs#L50 for an example

surjithctly commented:
@erezrokah There is already identity widget. See:  https://github.com/surjithctly/neat-starter/blob/master/admin/index.html#L13

I tried myself and was able to login. Something else missing? I'm not sure.

erezrokah commented:
> I tried myself and was able to login. Something else missing? I'm not sure.

It should be added to the root path of the site to handle invite links as those redirect there by default.
See more here https://community.netlify.com/t/common-issue-netlify-cms-git-gateway-email-not-confirmed/10690

surjithctly commented:
Ah.. Got it now. Added :) 

Are we good to go now?

erezrokah commented:
> Are we good to go now?

Let me test the flow one more time. I'll try to finalise it today.

erezrokah commented:
I'm getting a 404 on `config.yml` after deploying:
![image](https://user-images.githubusercontent.com/26760571/95760238-71089800-0cb3-11eb-9ae6-b259db0edae6.png)

See https://silly-cray-52b28f.netlify.app/admin/config.yml

surjithctly commented:
Okay. I was making one quick change before you test it :) 

All good now. I'll be not editing again before your review finishes,

Please do a one more final test :)
